### DISCUSSION TITLE
Editable overlay text via MQTT?

### DISCUSSION THREAD

**Discussion by Minglarn** [blakeblackshear/frigate#1849] [Ideas]
One thing i miss is a function for editable overlay text per camera. 
With my HikVisions I used to update tex overlay text with the actual temperature where the cam was pointing. 

Would this be doable via MQTT?

**Comment by tve** [comment]
If you're using a camera that streams h264 then frigate goes out of its way not to re-encode the stream, so what gets saved is what the camera sent. There's no stage where frigate could add any overlay. The only thing you could do is to specify input_args that make ffmpeg decode the video, add an overlay, and then re-encode. That wouldn't solve your use-case, though.

**Reply by Minglarn** [reply]
Okay... Well that makes it harder to implement. :(
But, how does Frigate add the timestamp? Isn't that some way to overlay text?

**Reply by blakeblackshear** [reply]
It's only added in the live view and snapshots because those feeds are decoded. RTMP and recordings are a direct copy and don't support the timestamp feature.
